Backend/editor/data/jsonConfig.go

```diff
@@ -0,0 +1,55 @@
+package data
+
+import (
+	"errors"
+	"reflect"
+	"strconv"
+
+	"cms.csesoc.unsw.edu.au/editor/data/datamodels/cmsmodel"
+	"cms.csesoc.unsw.edu.au/internal/cmsjson"
+)
+
+// models contains all the data models for the editor
+// this is the configuration required for the cmsjson module
+// cmsjson is a custom marshaller/unmarshaller that supports interface types
+// cmsjson works with arbtirary schemas so this model can be changed on a whim
+// note that cmsjson does not check that the provided types implement the interface
+// so please check that everything works prior to running the CMS
+var cmsJsonConf = cmsjson.Configuration{
+	// Registration for cmsmodel, when the LP is finally merged with CSESoc Projects
+	// this will also contain the registration for their data models
+	RegisteredTypes: map[reflect.Type]map[string]reflect.Type{
+		reflect.TypeOf((*cmsmodel.Component)(nil)).Elem(): {
+			"image":     reflect.TypeOf(cmsmodel.Image{}),
+			"paragraph": reflect.TypeOf(cmsmodel.Paragraph{}),
+		},
+
+		reflect.TypeOf((*Payload)(nil)).Elem(): {
+			"textEdit":  reflect.TypeOf(TextEdit{}),
+			"keyEdit":   reflect.TypeOf(KeyEdit{}),
+			"arrayEdit": reflect.TypeOf(ArrayEdit{}),
+		},
+	},
+}
+
+// small helper function to parse a JSON value of a specific type
+func parseDataGivenType(dataStr string, dataType string) (interface{}, error) {
```

Comment:
Not sure when this function is used? Would it be be better to just use a function in the `unmarshall.go` file by making that function public, or is this for something different / we need this function because inputs are different?

Reply:
I'm not sure either, Brian had it there initially so I'm assuming he plans on using it soon.

Reply:
This function was made on the original specification of taking in requests from frontend. Each request would pass in the payload with content and content type. Not sure if things have changed since I made this months ago?

Reply:
It's not currently in use, but was meant to be linked to the entrypoint once it was done. @Varun-Sethu thoughts?